source/features/quick-comment-edit.tsx

```diff
@@ -7,13 +7,18 @@ import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 
-function canEditDiscussion(): boolean {
-	return pageDetect.isDiscussion() && select.exists('[title^="You are a maintainer"], [title^="You are a collaborator"]');
+function canEditEveryComment(): boolean {
+	return select.exists([
+		'[aria-label^="You have been invited to collaborate"]',
+		'[aria-label^="You are the owner"]',
+		'[title^="You are a maintainer"]',
+		'[title^="You are a collaborator"]',
+	]) && select.exists('div.js-previewable-comment-form');
```

Comment:
Is this actually related to `canEditEveryComment`? This looks like a generic test to check whether there are any editable comments at all

Reply:
The `select.exists('div.js-previewable-comment-form')` ensures they can edit the first comment, and the rest checks if they are one of `owner`, `collaborator`, or `invited-to-collaborate`.

I'm not sure why `select.exists('div.js-previewable-comment-form')` exists, since if you are `owner`, `collaborator`, or `invited-to-collaborate`, you should be able to edit all, but it works well with this condition now.

Reply:
Hmmm you're right, the `select.exists('div.js-previewable-comment-form')` is redundant. I added it as a fix to https://github.com/sindresorhus/refined-github/pull/4749#issuecomment-915767672, but now I realise it doesn't even fix the issue. A better fix would be detecting the settings tab.

Reply:
I changed it now, let me know what you think